Public/Src/Utilities/Collections/ConcurrentBigMapEntry.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Utilities.Collections
+{
+    /// <summary>
+    /// This is a key value pair for the concurrent big map.
+    /// </summary>
+    /// <remarks>
+    /// We can't use the built-in KeyValuePair because it does not implement GetHashCode and Equals
+    /// </remarks>
+    public struct ConcurrentBigMapEntry<TKey, TValue> : IEquatable<ConcurrentBigMapEntry<TKey, TValue>>
+    {
+        /// <nodoc />
+        public ConcurrentBigMapEntry(TKey key, TValue value)
+        {
+            Key = key;
+            Value = value;
+        }
+
+        /// <nodoc />
+        public TKey Key { get; }
+
+        /// <nodoc />
+        public TValue Value { get; }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            return obj is KeyValuePair<TKey, TValue> pair &&

```

Comment:
```suggestion
            return obj is KeyValuePair<TKey, TValue> pair && Equals(pair);
```

Reply:
Actually this should have been `StructUtilities.Equals` ...